presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConnectorModule.java

```
@@ -0,0 +1,152 @@
+        binder.bind(ElasticsearchRecordSetProvider.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(ElasticsearchConnectorConfig.class);
+
+        jsonBinder(binder).addDeserializerBinding(Type.class).to(TypeDeserializer.class);
+        jsonCodecBinder(binder).bindJsonCodec(ElasticsearchTableDescription.class);
+
+        binder.install(new DecoderModule());
+    }
+
+    @Singleton
+    @Provides
+    public static ElasticsearchClient createElasticsearchClient(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> tableDescriptionCodec)
+    {
+        ImmutableMap.Builder<SchemaTableName, ElasticsearchTableDescription> builder = ImmutableMap.builder();
+        try {
+            for (File file : listFiles(config.getTableDescriptionDir())) {
+                if (file.isFile() && file.getName().endsWith(".json")) {
+                    ElasticsearchTableDescription table = tableDescriptionCodec.fromJson(readAllBytes(file.toPath()));
+                    String schemaName = firstNonNull(table.getSchemaName(), config.getDefaultSchema());
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, ElasticsearchTableDescription> tableDefinitions = builder.build();
+
+            builder = ImmutableMap.builder();
+            for (String definedTable : config.getTableNames()) {
+                SchemaTableName tableName;
+                try {
+                    tableName = parseTableName(definedTable);
+                }
+                catch (IllegalArgumentException iae) {
+                    tableName = new SchemaTableName(config.getDefaultSchema(), definedTable);
+                }
+
+                if (!tableDefinitions.containsKey(tableName)) {
+                    throw new IllegalArgumentException("Missing table definition for: " + tableName);
```

IIUC this exception will be thrown when the system is configured incorrectly. If that's the case it would be better to throw something like an `InvalidConfigurationException`.

---

will fix